core/src/main/scala/org/http4s/QValue.scala

```diff
@@ -98,10 +101,29 @@ object QValue {
   def unsafeFromString(s: String): QValue =
     fromString(s).fold(throw _, identity)
 
+  private[http4s] val parser: Parser[QValue] = {
+    import cats.parse.Parser.{char => ch, _}
+    import cats.parse.Rfc5234._
+
+    val optWS = (crlf.rep.with1 ~ wsp.rep1).rep
+
+    val qValue = string1(ch('0') ~ (ch('.') ~ digit.rep1).rep)
+      .map(
+        QValue
+          .fromString(_)
+          .valueOr(e => throw e.copy(sanitized = "Invalid q-value"))
+      )
+      .orElse1(
+        ch('1') *> (ch('.') *> ch('0').rep).rep.as(One)
```

Comment:
What's the second `.rep` for?

Reply:
The second `.rep` is to make a fractional part ".000" optional. Not sure how to do this properly in cats-parse

Reply:
Oh, right. It's `.?`.

Reply:
Thanks. Done